packages/ui/src/components/vuestic-components/va-pagination/VaPagination.vue

```diff
@@ -1,215 +1,258 @@
+  color: { type: String, default: 'primary' },
+})
+
+const mixinsArr = [
+  ContextPluginMixin,
+  StatefulMixin,
+  ColorThemeMixin,
+  paginationPropsMixin,
+]
+@Component({
   name: 'VaPagination',
   components: {
     VaButtonGroup,
     VaButton,
+    VaInput,
   },
-  mixins: [ContextPluginMixin],
-  props: {
-    value: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'value', 1)
-      },
-    },
-    visiblePages: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'visiblePages', 5)
-      },
-    },
-    pages: {
-      type: Number,
-      default () {
-        return getContextPropValue(this, 'pages', null)
-      },
-      required: true,
-    },
-    color: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'color', 'info')
-      },
-    },
-    disabled: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'disabled', false)
-      },
-    },
-    size: {
-      type: String,
-      default () {
-        return getContextPropValue(this, 'size', 'medium')
-      },
-      validator: value => {
-        return ['medium', 'small', 'large'].includes(value)
-      },
-    },
-    boundaryLinks: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'boundaryLinks', true)
-      },
-    },
-    directionLinks: {
-      type: Boolean,
-      default () {
-        return getContextPropValue(this, 'directionLinks', true)
-      },
-    },
-    iconFont: {
-      type: Object,
-      default () {
-        return getContextPropValue(this, 'iconFont', null)
-      },
-    },
-    iconFontRight: {
-      type: Object,
-      default () {
-        return getContextPropValue(this, 'iconFontRight', null)
-      },
-    },
-  },
-  data () {
-    return {
-      defaultIconClass: {
-        direction: 'chevron_left',
-        boundary: 'first_page',
-      },
-      defaultIconRightClass: {
-        direction: 'chevron_right',
-        boundary: 'last_page',
-      },
+})
+export default class VaPagination extends mixins(...mixinsArr) {
+  private defaultIconClass: IconSet = {
+    direction: 'chevron_left',
+    boundary: 'first_page',
+  }
+
+  private defaultIconRightClass: IconSet = {
+    direction: 'chevron_right',
+    boundary: 'last_page',
+  }
+
+  private inputValue = ''
+
+  private useTotal = false
+
+  private get iconClass () {
+    return Object.assign({}, this.defaultIconClass, (this as any).iconSet)
+  }
+
+  private get iconRightClass () {
+    return Object.assign({}, this.defaultIconRightClass, (this as any).iconSetRight)
+  }
+
+  private get lastPage () {
+    const { total, pageSize, pages } = (this as any)
+    return this.useTotal
+      ? Math.ceil(total / pageSize)
+      : pages
+  }
+
+  private get paginationRange () {
+    const { valueComputed, visiblePages, total, pageSize, boundaryNumbers, pages } = (this as any)
+    return this.useTotal
```

Comment:
For total/pages we may want to:
* add error for case when user specifies both (as that shouldn't be possible).
* for total mode - value should be items, not pages.